Plugin/Model/Order/ItemRepositoryPlugin.php

```diff
@@ -0,0 +1,166 @@
+
+    /**
+     * @var GetDefaultSourceSelectionAlgorithmCodeInterface
+     */
+    private $getDefaultSourceSelectionAlgorithmCode;
+
+    /**
+     * @var SourceSelectionServiceInterface
+     */
+    private $sourceSelectionService;
+
+    /**
+     * ItemRepositoryPlugin constructor.
+     *
+     * @param OrderItemExtensionFactory $orderItemExtensionFactory
+     * @param LoggerInterface $logger
+     * @param Order $order
+     * @param GetSourceSelectionResultFromOrder $sourceSelectionResult
+     * @param GetInventoryRequestFromOrder $getInventoryRequestFromOrder
+     * @param GetDefaultSourceSelectionAlgorithmCodeInterface $getDefaultSourceSelectionAlgorithmCode
+     * @param SourceSelectionServiceInterface $sourceSelectionService
+     */
+    public function __construct(
+        OrderItemExtensionFactory $orderItemExtensionFactory,
+        LoggerInterface $logger,
+        Order $order,
+        GetSourceSelectionResultFromOrder $sourceSelectionResult,
+        GetInventoryRequestFromOrder $getInventoryRequestFromOrder,
+        GetDefaultSourceSelectionAlgorithmCodeInterface $getDefaultSourceSelectionAlgorithmCode,
+        SourceSelectionServiceInterface $sourceSelectionService
+    ) {
+        $this->orderItemExtensionFactory = $orderItemExtensionFactory;
+        $this->logger = $logger;
+        $this->order = $order;
+        $this->sourceSelectionResult = $sourceSelectionResult;
+        $this->getInventoryRequestFromOrder = $getInventoryRequestFromOrder;
+        $this->getDefaultSourceSelectionAlgorithmCode = $getDefaultSourceSelectionAlgorithmCode;
+        $this->sourceSelectionService = $sourceSelectionService;
+    }
+
+    /**
+     * Set additional attributes when requesting a single order item
+     *
+     * @param ItemRepository  $subject
+     * @param OrderItemInterface|Item $result
+     * @return OrderItemInterface|Item
+     */
+    public function afterGet(ItemRepository $subject, OrderItemInterface $result)
+    {
+        return $this->applyExtensionAttributesToOrderItem($result, $result);
+    }
+
+    /**
+     * Set additional attributes when requesting a list of order items
+     *
+     * @param ItemRepository $subject
+     * @param OrderItemSearchResultInterface $result
+     * @return OrderItemSearchResultInterface
+     */
+    public function afterGetList(ItemRepository $subject, OrderItemSearchResultInterface $result)
+    {
+        foreach ($result->getItems() as $item) {
+            $this->applyExtensionAttributesToOrderItem($item, $result->getItems());
+        }
+
+        return $result;
+    }
+
+    /**
+     * Apply extension attributes to order item
+     *
+     * @param OrderItemInterface $orderItem
+     * @param array $allItems
+     * @return OrderItemInterface
+     */
+    private function applyExtensionAttributesToOrderItem(OrderItemInterface $orderItem, array $allItems) : OrderItemInterface
+    {
+        $orderProduct = $orderItem->getProduct();
```

Comment:
I don't think you're using the product in your functionality? This was originally implemented for a specific use case on Seraphine, it's not needed in this module